crates/engine/tree/src/tree/payload_processor/multiproof.rs

```diff
@@ -1146,6 +1168,58 @@ impl MultiProofTask {
 
                 false
             }
+            // Process Block Access List (BAL) - complete state changes provided upfront
+            MultiProofMessage::BlockAccessList(bal) => {
+                trace!(target: "engine::tree::payload_processor::multiproof", "processing MultiProofMessage::BAL");
+
+                if ctx.first_update_time.is_none() {
+                    self.metrics
+                        .first_update_wait_time_histogram
+                        .record(ctx.start.elapsed().as_secs_f64());
+                    ctx.first_update_time = Some(Instant::now());
+                    debug!(target: "engine::tree::payload_processor::multiproof", "Started state root calculation from BAL");
+                }
+
+                // Convert BAL to HashedPostState and process it
+                match bal_to_hashed_post_state(&bal, &provider) {
+                    Ok(hashed_state) => {
+                        debug!(
+                            target: "engine::tree::payload_processor::multiproof",
+                            accounts = hashed_state.accounts.len(),
+                            storages = hashed_state.storages.len(),
+                            "Processing BAL state update"
+                        );
+
+                        // Use Transaction(0) as source for BAL-derived state updates
+                        batch_metrics.state_update_proofs_requested += self.on_hashed_state_update(
+                            StateChangeSource::Transaction(0),
```

Comment:
would be nice to identify BAL state updates separately, as IIUC this `source` is only used for logging https://github.com/paradigmxyz/reth/blob/3c4740dab648b7f86ca814cce02a36bd5bfbeade/crates/engine/tree/src/tree/payload_processor/multiproof.rs#L472-L480